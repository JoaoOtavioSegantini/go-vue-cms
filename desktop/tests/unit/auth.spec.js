import Auth from "@/components/Auth.vue";
import { mount } from "@vue/test-utils";

// const sinon = require('sinon');

// const mockRequest = () => {
//   return {
//     users: []
//   };
// };

// const mockResponse = () => {
//   const res = {};
//   res.status = sinon.stub().returns(res);
//   res.json = sinon.stub().returns(res);
//   return res;
// };

window.axios = {
  post: jest
    .fn(() => {
      return {
        then: jest.fn(() => "your faked response"),
      };
    })
    .mockName("axiosPost"),
};

describe("Auth.vue", () => {
  afterEach(() => {
    jest.restoreAllMocks();
  });

  function factory() {
    return mount(Auth, {
      mocks: {
        axios: window.axios,
      },
      data() {
        return {
          user: {
            email: "acbdfg@gmail.com",
            password: "987654",
          },
        };
      },
    });
  }

  it("renders props when passed", async () => {
    const msg = "Autenticação Informe suas credenciaisEmailSenhaAcessar";

    const wrapper = factory();

    console.log(wrapper.vm);
    wrapper.findAll("input")[0].setValue("acbdfg@gmail.com");
    wrapper.findAll("input")[1].setValue("987654");

    await wrapper.find("form").trigger("submit");
    console.log(wrapper.emitted());

    await wrapper.vm.auth();

    expect(window.axios.post).toBeCalledWith(
      "/api/token",
      { email: "acbdfg@gmail.com", password: "987654" },
      {
        swalMessage: "Você acessou com sucesso",
        swalTitle: "Autenticado com sucesso",
      }
    );
    expect(wrapper.text()).toMatch(msg);
    expect(wrapper.html()).toMatchSnapshot();
  });
});
