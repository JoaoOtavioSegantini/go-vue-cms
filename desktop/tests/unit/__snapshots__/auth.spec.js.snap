// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Auth.vue renders props when passed 1`] = `
"<div class=\\"container-fluid\\">
  <div class=\\"row\\">
    <div class=\\"col\\">
      <div class=\\"card\\">
        <div class=\\"card-body\\">
          <h5 class=\\"title\\"><i class=\\"fa fa-file-text-o\\"></i> Autenticação <small>Informe suas credenciais</small></h5>
          <form>
            <div class=\\"form-group\\"><label for=\\"email\\">Email</label><input type=\\"email\\" class=\\"form-control\\" id=\\"email\\" placeholder=\\"Email...\\"></div>
            <div class=\\"form-group\\"><label for=\\"password\\">Senha</label><input type=\\"password\\" class=\\"form-control\\" id=\\"password\\" placeholder=\\"Sua senha\\"></div><button type=\\"submit\\" class=\\"btn btn-primary\\">Acessar</button>
          </form>
        </div>
      </div>
    </div>
  </div>
</div>"
`;
